test/topology_custom/test_lwt_semaphore.py

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import time
+from test.pylib.rest_client import inject_error
+from test.pylib.util import wait_for_cql_and_get_hosts
+import pytest
+from cassandra.protocol import WriteTimeout
+
+@pytest.mark.asyncio
+async def test_cas_semaphore(manager):
+    servers = await manager.servers_add(3, cmdline=['--smp', '1', '--write-request-timeout-in-ms', '500'])
```

Comment:
Does it need 3 servers? IIUC even 1 server should be enough (unless we optimize Paxos away with RF=1?)

Reply:
with RF=1 it may complete the operation without preemption. Well, may be in debug is will be enough, but better to work in dev as well.

Reply:
Why would it complete without preemption?

Reply:
Why not? Where would it preempt? In order to trigger the bug we need one query to preempt after prepare and another query to do a prepare with higher ballot.

Reply:
I tried to run with one server (and one small fix to make it work) and strangely it is a debug build that becomes flaky without a fix (does not reproduce the problem while it should sometimes). So may be we can run with one server and hope that dev build will remain to be non fluky.

Reply:
Ok, let's  keep 3 so it's not flaky (although 2 would also suffice probably)

Reply:
I actually just pushed a new version with one and also changing the constructor to default. The test is fluky without the fix, so in case of a regression it may become fluky, but it should be enough to catch the regression.

Reply:
OK